service/indexer.py

```diff
@@ -0,0 +1,110 @@
+import sys
+import asyncio
+from .schema import validate_schema
+
+
+class index_worker:
+    """
+        Index worker for queue(TBA), including:
+        - put items on queue
+        - get items on queue
+        - drain items on queue
+        - get positional items of queue
+    """
+
+    def __init__(self, queue):
+        self.queue = queue
+
+    def put_item(self, item):
+        """
+        Put item on queue
+        """
+        return self.queue.put(item)
+
+    def get_message(self):
+        """
+        Get one item from queue
+        """
+        queue_item = self.queue.pop()
+        if validate_schema(queue_item):
+            return queue_item
+
+    def fetch_index(self, index: int):
+        """
+        Fetch index of queue
+        """
+        return self.queue[index]
+
+    def drain_queue(self):
+        """
+        Drain queue
+        """
+
+        return self.queue.drain()
+
+    def message_recieved(self, item):
+        """
+
+        """
+        # check is one message recieved
+        if item:
+            return True
+
+
+
+async def indexer_engine(worker=None, sleep_time=None):
+    """
+        Engine of the indexer where tasks ran until complete, includes:
+        - worker: queue and its tasks
+        - sleep_time: amount to wait for an asyncio task
+    """
+    while True:
+        try:
+            await asyncio.sleep(sleep_time)
```

Comment:
we might need to talk to Rob but I'm pretty sure the result of our last design session is we want non async and one message at a time to dodge worrying about multiple deliveries.